demosplan/DemosPlanCoreBundle/Logic/OwnsProcedureConditionFactory.php

```diff
@@ -135,11 +169,91 @@ public function hasPlanningAgencyRole(): FunctionInterface
     /**
      * @return FunctionInterface<bool>
      */
-    public function isInCustomer(Customer $customer): FunctionInterface
+    protected function isUserInCustomer(Customer $customer): FunctionInterface
+    {
+        $customerId = $customer->getId();
+        Assert::notNull($customerId);
+
+        if ($this->userOrProcedure instanceof User) {
+            $user = $this->userOrProcedure;
+
+            return $user->isConnectedToCustomerId($customerId)
+                ? $this->conditionFactory->true()
+                : $this->conditionFactory->false();
+        }
+
+        return $this->conditionFactory->propertyHasValue($customer->getId(), ['roleInCustomers', 'customer', 'id']);
+    }
+
+    /**
+     * @return ClauseFunctionInterface<bool>
+     */
+    public function isEitherTemplateOrProcedure(bool $template): ClauseFunctionInterface
     {
-        return $this->conditionFactory->propertyHasValue(
-            $customer->getId(),
-            ['roleInCustomers', 'customer', 'id']
-        );
+        if ($this->userOrProcedure instanceof User) {
+            return $this->conditionFactory->propertyHasValue($template, ['master']);
+        }
+
+        $procedure = $this->userOrProcedure;
+
+        return $procedure->getMaster() === $template
+            ? $this->conditionFactory->true()
+            : $this->conditionFactory->false();
+    }
+
+    /**
+     * @return ClauseFunctionInterface<bool>
+     *
+     * @throws PathException
+     */
+    public function userIsExplicitlyAuthorized(): ClauseFunctionInterface
+    {
+        if ($this->userOrProcedure instanceof User) {
+            $user = $this->userOrProcedure;
+
+            return $this->conditionFactory->propertyHasStringAsMember($user->getId(), ['authorizedUsers']);
+        }
+
+        $procedure = $this->userOrProcedure;
+
+        return $this->conditionFactory->propertyHasAnyOfValues($procedure->getAuthorizedUserIds(), ['id']);
+    }
+
+    /**
+     * Users that are in the same organisation as the one of the user that created a procedure, own that procedure.
+     *
+     * @return ClauseFunctionInterface<bool>
+     *
+     * @throws PathException
+     */
+    public function userOwnsProcedureViaOrgaOfUserThatCreatedTheProcedure(): ClauseFunctionInterface
+    {
+        if ($this->userOrProcedure instanceof User) {
+            $user = $this->userOrProcedure;
+
+            return $this->conditionFactory->propertyHasValue($user->getOrganisationId(), ['orga']);
+        }
+
+        $procedure = $this->userOrProcedure;
+
+        return $this->conditionFactory->propertyHasValue($procedure->getOrgaId(), ['orga', 'id']);
+    }
+
+    /**
+     * @return ClauseFunctionInterface<bool>
+     *
+     * @throws PathException
+     */
+    public function isUndeletedProcedure(): ClauseFunctionInterface
```

Comment:
Undeleted sounds a bit strange here. Perhaps `isNotDeleted`?

Reply:
I renamed it to `isNotDeletedProcedure`, to keep the clarity that this checks the deletion of the procedure, not the user. e54af829944d655083c2a208ead4fcc931ff6bbd